docs/docs-beta/docs/guides/deployment/docker.md

```diff
@@ -1,3 +1,197 @@
 ---
-title: "Deploying to Docker"
+title: "Deploying with Docker"
+description: A guide to deploying Dagster with Docker.
 ---
+
+This guide provides instructions for deploying Dagster using Docker. It's common to use Docker containers to run the Dagster web server, daemon, and code servers. These containers are typically deployed using AWS ECS or another container-based orchestration system.
```

Comment:
In this guide it seems like the intention is the Docker containers are used exclusively through docker-compose, something that might not translate well to ECS (please correct me if I'm wrong).

Should we set that expectation, that this guide is for running multiple containers on a single compute instance like EC2?

Reply:
Ah this is a great point. @gibsondan @jmsanders – curious if you have instincts here? Is running multiple containers on a single compute instance a common enough pattern to warrant a guide?

Reply:
I think the docker compose file is useful because it’s a concise way to see all the moving parts in one place. In some sense, everything else (K8s, ECS) is just an implementation of the compose file (albeit with a lot more bells and whistles). In fact, there are tools out there that let you deploy from a docker compose file to other container orchestrators. 

I have no opinion on if that deserves its own guide or should be treated more as reference material.

Reply:
Ok I'm going to rename this guide to "Deploying Dagster with Docker Compose"